src/detect-modbus.c

```diff
@@ -570,442 +171,179 @@ void DetectModbusRegister(void)
-    DetectEngineCtxFree(de_ctx);
-    PASS;
-}
-
-/** \test Signature containing an unit_id and a read access at an address. */
-static int DetectModbusTest12(void)
-{
-    DetectEngineCtx     *de_ctx = NULL;
-    DetectModbus        *modbus = NULL;
-    DetectModbusMode    mode = DETECT_MODBUS_EQ;
-
-    uint8_t type = MODBUS_TYP_READ;
-
-    de_ctx = DetectEngineCtxInit();
-    FAIL_IF_NULL(de_ctx);
-
-    de_ctx->flags |= DE_QUIET;
-
-    de_ctx->sig_list = SigInit(de_ctx, "alert modbus any any -> any any "
-                                       "(msg:\"Testing modbus.access\"; "
-                                       "modbus: unit 10, access read, address 1000;  sid:1;)");
-    FAIL_IF_NULL(de_ctx->sig_list);
-    FAIL_IF_NULL(de_ctx->sig_list->sm_lists_tail[g_modbus_buffer_id]);
-    FAIL_IF_NULL(de_ctx->sig_list->sm_lists_tail[g_modbus_buffer_id]->ctx);
-
-    modbus = (DetectModbus *) de_ctx->sig_list->sm_lists_tail[g_modbus_buffer_id]->ctx;
-
-    FAIL_IF_NOT((*modbus->unit_id).min == 10);
-    FAIL_IF_NOT((*modbus->unit_id).mode == mode);
-    FAIL_IF_NOT(modbus->type == type);
-    FAIL_IF_NOT((*modbus->address).mode == mode);
-    FAIL_IF_NOT((*modbus->address).min == 1000);
+    FAIL_IF_NOT(PacketAlertCheck(p, 1));
 
+    AppLayerParserThreadCtxFree(alp_tctx);
+    DetectEngineThreadCtxDeinit(&tv, det_ctx);
     SigGroupCleanup(de_ctx);
-    SigCleanSignatures(de_ctx);
     DetectEngineCtxFree(de_ctx);
-    PASS;
-}
-
-/** \test Signature containing a range of unit_id. */
-static int DetectModbusTest13(void)
-{
-    DetectEngineCtx     *de_ctx = NULL;
-    DetectModbus        *modbus = NULL;
-    DetectModbusMode    mode = DETECT_MODBUS_RA;
 
-    de_ctx = DetectEngineCtxInit();
-    FAIL_IF_NULL(de_ctx);
-
-    de_ctx->flags |= DE_QUIET;
-
-    de_ctx->sig_list = SigInit(de_ctx, "alert modbus any any -> any any "
-                                       "(msg:\"Testing modbus.access\"; "
-                                       "modbus: unit 10<>500;  sid:1;)");
-    FAIL_IF_NULL(de_ctx->sig_list);
-    FAIL_IF_NULL(de_ctx->sig_list->sm_lists_tail[g_modbus_buffer_id]);
-    FAIL_IF_NULL(de_ctx->sig_list->sm_lists_tail[g_modbus_buffer_id]->ctx);
-
-    modbus = (DetectModbus *) de_ctx->sig_list->sm_lists_tail[g_modbus_buffer_id]->ctx;
-
-    FAIL_IF_NOT((*modbus->unit_id).min == 10);
-    FAIL_IF_NOT((*modbus->unit_id).max == 500);
-    FAIL_IF_NOT((*modbus->unit_id).mode == mode);
-
-    SigGroupCleanup(de_ctx);
-    SigCleanSignatures(de_ctx);
-    DetectEngineCtxFree(de_ctx);
+    StreamTcpFreeConfig(TRUE);
+    FLOW_DESTROY(&f);
+    UTHFreePacket(p);
     PASS;
 }
 
 /**
  * \brief this function registers unit tests for DetectModbus
  */
-void DetectModbusRegisterTests(void)
+static void DetectModbusRegisterTests(void)
 {
-    UtRegisterTest("DetectModbusTest01 - Testing function",
-                   DetectModbusTest01);
-    UtRegisterTest("DetectModbusTest02 - Testing function and subfunction",
```

Comment:
Why did you remove these unit tests ?
They should be useful to check if we still have the same behavior, right ?

Reply:
Maybe they can be converted to rust tests? (`cargo test` style)

Reply:
All of these tests, and more, are now part of the rust module. Is there benefit to adding more tests to this C module?

See [rust/src/modbus/detect.rs](https://github.com/cccs-sadugas/suricata/blob/rust-modbus-v2/rust/src/modbus/detect.rs#L492).

Overall our testing strategy is as follows:
- tests related to the parsing of data are in `sawp-modbus` (81% coverage)
- tests related to parsing keywords and other suricata functionality are in their respective rust module (not sure these are recognized by codecov?)
- tests related to integration are in the C wrapper (minimal, to avoid duplication with unit tests)
- system tests are in suricata-verify
- we are currently soak testing on a staging environment to see if there are any defects

Reply:
>  Is there benefit to adding more tests to this C module?

I did not mean to add more C tests.
Translating the C tests to Rust is ok :-)

But I would like it (the moving of C tests ro Rust) to be in its own commit after the parser itself went from C to Rust.
This way, we can check easily that the translation of the parser from C to Rust did not break these tests.
When I translated SSH from C to Rust, these C tests helped me get some edge cases right

Reply:
> * tests related to parsing keywords and other suricata functionality are in their respective rust module (not sure these are recognized by codecov?)

No, our codecov integration doesn't support Rust yet. Still need to look into that.